File: Easy/nCr.cpp
//{ Driver Code Starts
//Initial template for c++

#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
//User function Template for C++

class Solution
{   
public:
     long pot(long a, long y)
     { 
        int m = pow(10,9)+7;
        if(y==0)    return 1;
        long p = pot(a,y/2)%m;
        p = (p*p)%m;
        return (y%2==0)?p:(a*p)%m;
    }
    
    long int ncr(int n, int r)
    {
        //complete the function here
        int m = pow(10,9)+7;
        
        long dp[n+1];
        dp[0] = 1;
        
        for(int i = 1; i<=n; i++) dp[i] = (i*dp[i-1])%m;

        if(r>n) return 0;

        long q = dp[n], down = (dp[r]*dp[n-r])%m;
        long ans = (q*pot(down, m-2))%m;
        return (int) ans;
    }
};

//{ Driver Code Starts.

int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int n, r;
        cin >> n >> r;
        Solution ob;
        cout << ob.ncr(n, r) << endl;
    
    }
    return 0; 
} 

// } Driver Code Ends

File: Medium/nCr.py
#User function Template for python3

class Solution:
    def nCr(self, n, r):
        # code here
        # n! / r!(n-r)!
        m = 1000000007
        if r > n:
            return 0
        
        elif r == 1:
            return 1
        
        else:
            a = 1
            for i in range(1, r + 1):
                a *= n
                n -= 1
        
            b = 1
            for i in range(1, r + 1):
                b *= i 
        
        return (a // b) % m
#{ 
 # Driver Code Starts
#Initial Template for Python 3

import sys
sys.setrecursionlimit(10**6)

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n, r = [int(x) for x in input().split()]
        
        ob = Solution()
        print(ob.nCr(n, r))
# } Driver Code Ends